### DISCUSSION TITLE
What is the purpose of rocket_db_pools?

### DISCUSSION THREAD

**Discussion by fenhl** [rwf2/Rocket#2175] [Questions] [answered]
The latest release candidate introduced the [`rocket_db_pools`](https://docs.rs/rocket_db_pools) crate, which provides database connections. How does this differ from calling [`Rocket::manage`](https://docs.rs/rocket/0.5.0-rc.2/rocket/struct.Rocket.html#method.manage) with a [`sqlx::PgPool`](https://docs.rs/sqlx/0.5.13/sqlx/type.PgPool.html)? Should I switch to it? Why/why not?

**Comment by SergioBenitez** [comment, marked as answer]
Rocket already had database pooling support. This crate is for async database connections. You should absolutely use it instead of doing it yourself. 

What does it do that doing it yourself wouldn't? Reading through the docs should answer that: https://docs.rs/rocket_db_pools/0.1.0-rc.2/rocket_db_pools/. In short:

- Makes it trivial to configure databases, and to do so correctly. 
- Makes it trivial to use more than one database at a time, especially those of the same type. You can't manage two PgPools without extra effort.
- Properly handles all error cases for you automatically.
- Properly configures databases according to other Rocket configuration.
- Significantly decreases the amount of code you need to write to connect to databases.
- Handles vetting database compatibility with Rocket and other parts of the ecosystem.  
- Applies best practice, and at times robustness-necessitated, database and ORM configuration settings automatically.
- Probably other things.
